### ISSUE TITLE
Normalize and merge copyright lines

### ISSUE THREAD

**Issue by kellnerd** [kellnerd/harmony#23]
Continuing the discussion from https://github.com/kellnerd/harmony/issues/22#issuecomment-2156460070

> We can factor out the copyright normalization logic and reuse it for other providers, e.g. as suggested for Tidal in https://community.metabrainz.org/t/harmony-music-metadata-aggregator-and-musicbrainz-importer/698641/15
> But I'd do this after merging this PR. It also needs some further research. I know Tidal includes the copyright text both with and without the © symbol. What I'm unsure is whether this strictly contains copyright © info, or whether it also can sometimes contain phonographic copyright ℗ info. Spotify has those separated, which makes it easier.

I fully agree, this is enough for its own PR and it needs more research.
Tidal also has a `copyright` property at the track level by the way, this should also be considered if it is different from the release level coypright. So far they were identical for the releases which I have checked, maybe a compilation has different values there.

For starters I have a [commit](https://github.com/kellnerd/harmony/commit/36b8d28676c1e6ccb46469cbf0a6273dfda7f3b3) in the `dev` branch which displays the alternative copyright values.
When we have more examples we can decide how the release merge algorithm should handle these, one possibility would be to keep all and deduplicate them.

**Comment by phw** [comment]
For Tidal it is more complicated, the `copyright` field can contain both (P) or (C) entries, and it can be with or without a symbol (ASCII or proper character).

- This example has "(P) 2016 Century Media Records Ltd.": https://tidal.com/album/64944658
- Here it is "© 2012 S. Carter Enterprises, LLC.": https://tidal.com/browse/album/130201923
- And this is just "Nuclear Blast" without any symbol: https://tidal.com/browse/album/180584784

So I'm not sure whether we can or should add © or ℗ (or maybe "© + ℗") to the string if the symbol is missing. What we could do is converting the ASCII variants into symbols.

**Comment by phw** [comment]
I wonder that maybe "copyright" should be a list of strings per provider instead of a single string. Currently it is I think only Spotify which offers explicit distinct values for (P) and (C). But having those as actual separate values will make de-duplication easier. Right now the provider just separates both with a new line.

**Comment by SheamusPatt** [comment]
Why are you putting copyright / phonographic copyright in the annotation at all? There are release relationships where they can be put, which would link the release to the associated label or artist.
It would require matching the text after the statement to an artist or label (could be either).
That would apply to phonograpic copyright on tracks as well.

**Comment by arsinclair** [comment]
> Why are you putting copyright / phonographic copyright in the annotation at all? There are release relationships where they can be put, which would link the release to the associated label or artist.

So that this information is preserved, in case the release source page 'disappears'. There's no automated way to submit relationships to MB yet.

**Comment by yindesu** [comment]
Frankly, the information should be put in the edit note if it's just to preserve a snapshot of the data, as the annotation isn't the correct place for it anyway...

It's also frankly useful to preserve the differences between the copyright statements of each platform, as Harmony's default behavior makes errors that many users are unable to catch - example,

https://harmony.pulsewidth.org.uk/release?url=&gtin=602508485701&region=US&itunes=&spotify=&tidal=&qobuz= is only caught because the track counts differ.  But the normal case is that the track counts are the same, and Harmony proceeds to incorrectly merge two separate releases into one:
https://harmony.pulsewidth.org.uk/release?url=&gtin=602455674586&region=US&itunes=&spotify=&tidal=&qobuz=

In this last example, the Tidal copyright statement differs materially because the true release label is different from the other platforms.

**Comment by arsinclair** [comment]
> Frankly, the information should be put in the edit note if it's just to preserve a snapshot of the data, as the annotation isn't the correct place for it anyway...

I disagree with this bit, it's not a good idea to put it in the edit note only, as then this information is buried too deep. On the other hand, if it's in the annotation, someone might come later and enter it properly (I often do this).

[Annotation guidelines](https://musicbrainz.org/doc/Annotation) don't specifically forbid this. And as it allows some non-conformant metadata to be entered as annotation,

> Their purpose is to add information that usually doesn't fit into the strict structural data schema of MusicBrainz (be it due to technical limitations that may be addressed later, or because the information in itself has to be free-text)

I believe that a conformant metadata can be put there too, with the purpose of entering it later.
